tests/src/FunctionalJavascript/JavascriptBehavioursTest.php

```diff
@@ -196,23 +208,7 @@ public function testEclDatePicker(): void {
    */
   public function testContextNavPattern(): void {
     $this->drupalGet('/oe_theme_js_test/ui_patterns/context_nav');
-    $script = <<<EndOfScript
-(function(){
-if (typeof(window.collectedErrors) == 'undefined') {
-  return;
-}
-var result = '';
-window.collectedErrors.forEach(function(value) {
-   result += value.data + '[NLS]';
-});
-return result;
-})()
-EndOfScript;
-    $errors = $this->getSession()->evaluateScript($script);
-    if ($errors) {
-      throw new \RuntimeException('Javascript error: ' . str_replace('[NLS]', PHP_EOL, $errors));
-    }
-
+    $this->failOnJavascriptErrors();
```

Comment:
It catches errors related to context_nav. But what is about other tests? For example issues in js/ecl_datepicker.js can be caught by testEclDatePicker().
Might be we can implement something like:
```
  /**
   * @inheritdoc
   */
  protected function drupalGet($path, array $options = [], array $headers = []) {
    $out = parent::drupalGet($path, $options, $headers = []);

    $this->failOnJavascriptErrors();
    return $out;
  }
```
It will catch js errors in all tests in JavascriptBehavioursTest.

Reply:
Good idea. I've used overriding of this method.